### ISSUE TITLE
`space-in-generics`: add a rule to remove extra spaces in TypeScript generics

### ISSUE THREAD

**Issue by vnva** [eslint-stylistic/eslint-stylistic#649]
### Clear and concise description of the problem

The proposal is to add a rule that will remove spaces in TypeScript generics.

Examples:

```ts
interface Foo< T > {} // Bad
interface Foo<T> {} // Good

type Bar< T, U > = T & U; // Bad
type Bar<T, U> = T & U; // Good

function foo< T >(arg: T): T { return arg; } // Bad
function foo<T>(arg: T): T { return arg; } // Good


const x: Array < Array < string > > = []; // Bad
const x: Array<Array<string>> = []; // Good
```

### Suggested solution

Similar rule:

[@stylistic/space-in-parens](https://eslint.style/rules/default/space-in-parens)

My implementation:

[Rule](https://github.com/vnva/eslint-plugin/blob/main/src/rules/space-in-generics.ts)
[Tests](https://github.com/vnva/eslint-plugin/blob/main/src/rules/space-in-generics.test.ts)

### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### Contributes

- [X] If this feature request is accepted, I am willing to submit a PR to fix this issue

**Comment by 9romise** [comment]
similar to #480

**Comment by vnva** [comment]
> similar to #480

This issue is solved by [type-generic-spacing](https://eslint.style/rules/plus/type-generic-spacing). Perhaps this rule should be extended.

**Comment by vnva** [comment]
I'm not the only one

[[Is there any workaround / solution in Eslint for TypeScript to get spaces in generic classes/methods](https://stackoverflow.com/questions/67232000/is-there-any-workaround-solution-in-eslint-for-typescript-to-get-spaces-in-gen)](https://stackoverflow.com/questions/67232000/is-there-any-workaround-solution-in-eslint-for-typescript-to-get-spaces-in-gen/79314717#79314717)
